hyperhtml.js

```diff
@@ -278,15 +305,61 @@ var hyperHTML = (function () {'use strict';
   // Helpers
   // -------------------------
 
-  // it does exactly what it says
-  function appendNodes(node, childNodes) {
-    for (var
-      i = 0,
-      length = childNodes.length;
-      i < length; i++
-    ) {
-      node.appendChild(childNodes[i]);
+  // given a generic text or comment node
+  // remove the surrounding emptiness
+  function cleanAround(textNode, sibling) {
+    var adj = textNode[sibling];
+    if (adj && adj.nodeType === 3 && trim.call(adj.textContent).length < 1) {
+      textNode.parentNode.removeChild(adj);
+      cleanAround(textNode, sibling);
+    }
+  }
+
+  // find a live node through a virtual one
+  function find(live, virtual) {
+    var i, length, name, parentNode, map = [];
+    switch(virtual.nodeType) {
+      case 1:
```

Comment:
same as above, although I do appreciate your reviews and nitpicks and I'll try to wait for them next PR (I have to remember to ask you to review via GitHub, sorry I didn't this time)